core-domain/src/main/java/org/uniprot/core/uniprotkb/UniProtKBEntry.java

```diff
@@ -76,6 +77,8 @@
 
     EntryInactiveReason getInactiveReason();
 
+    Map<String, Object> getExtraAttributes();
```

Comment:
Can we also add a method  getExtraAttribute(ExtraAttributeName name)? fell free to give it a better name if you prefer...

Reply:
Added method getExtraAttributeValue(String).

Reply:
I am intentionally not passing the enum because then we will have to change the Map key too. And then we need to change the default behaviour of Json writer as well.

Reply:
Can you please do the improve implementation to use Enum?

Reply:
The way that it is, we could add whatever we want in the extended map. I  think initial idea was to add as few things as possible in this map, this is why adding the enum would help us to guarantee it. this is why I have suggested it…

Reply:
Based on my discussion with @jieluo321 , the main goal of adding a map is to avoid adding many fields(computed/derived) in UniProtEntry. EvenEven by adding enum, people can add whatever they want they just need to add one enum value.

Reply:
To make it consistent, add in the interface and builder or  remove it from builder to make consistent...

Reply:
Ok Leo. I have removed the enum from the builder. 
Can i merge now?

Reply:
Merging now. I can file a ticket if you need any other change.